### PR TITLE
add ResNeSt model

### PR DESCRIPTION
**Pull request by zhanghang1989**
|                        | crop size | Gluon |
|--------------|----------|-------|
| ResNeSt-50  | 224          | 81.04 |
| ResNeSt-101 | 256          | 82.81 |
| ResNeSt-200 | 320         | 83.88 |
| ResNeSt-269 | 416          | 84.53 |

### CODE REVIEW

File: gluoncv/nn/splat.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        inter_channels = max(in_channels*radix//2//r, 32)
+        self.radix = radix
+        self.cardinality = groups
+        self.conv = Conv2D(channels*radix, kernel_size, strides, padding, dilation,
+                           groups=groups*radix, *args, in_channels=in_channels, **kwargs)
+        if USE_BN:
+            self.bn = norm_layer(in_channels=channels*radix, **norm_kwargs)
+        self.relu = Activation('relu')
+        self.fc1 = Conv2D(inter_channels, 1, in_channels=channels, groups=self.cardinality)
+        if USE_BN:
```
Comment (zhreshold) [inline gluoncv/nn/splat.py:24]: maybe `if norm_layer is not None` is sufficient and more flexible?

File: scripts/classification/imagenet/train_horovod.py [hunk 2]
```diff
@@ -0,0 +1,543 @@
...
+import math
+import time
+import random
+import gluoncv as gcv
+gcv.utils.check_version('0.6.0')
+
+#from gluoncv.model_zoo import get_model
+from autogluon.model_zoo import get_model
+from gluoncv.utils import makedirs#, LRSequential, LRScheduler
+import horovod.mxnet as hvd
```
Comment (hetong007) [inline scripts/classification/imagenet/train_horovod.py:40]: Do we have a tutorial on setting up Horovod with GluonCV?

Comment (zhanghang1989) [reply scripts/classification/imagenet/train_horovod.py:40]: Probably not.

@Jerryzcn Do we have one for mask rcnn?

Comment (zhanghang1989) [reply scripts/classification/imagenet/train_horovod.py:40]: The horovod installation is pretty difficult though.

File: gluoncv/nn/splat.py (hunk 1 above)
Comment (zhanghang1989) [reply gluoncv/nn/splat.py]: Agree. Just changed it.

**Comment by zhanghang1989** [comment]
do we still need to support python 2 in CI @zhreshold

**Comment by zhanghang1989** [comment]
Hi  @zhreshold @hetong007 , I have added unit test, documentation and paper reference. Let me know if you have further comments

**Comment by mli** [comment]
Job PR-1247-14 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1247/14/index.html 
Code coverage of this PR:  vs. Master:

**Comment by zhanghang1989** [comment]
The training command is added to https://github.com/dmlc/web-data/pull/242

**Comment by mli** [comment]
Job PR-1247-15 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1247/15/index.html 
Code coverage of this PR:  vs. Master:

**Comment by zhanghang1989** [comment]
I have tested the training scripts. Everything is working properly. I am merging it now and will create a separate PR for semantic segmentation models using ResNeSt.

**Pull request merged**